pyccel/codegen/printing/pycode.py

```diff
@@ -153,7 +153,10 @@ def _print_Variable(self, expr):
 
     def _print_FunctionDefArgument(self, expr):
         if expr.has_default:
-            return '{} = {}'.format(self._print(expr.name), self._print(expr.value))
+            if isinstance(expr.value, FunctionDef):
+                return '{} = {}'.format(self._print(expr.name), self._print(expr.value.name))
+            else:
+                return '{} = {}'.format(self._print(expr.name), self._print(expr.value))
```

Comment:
I am not sure that I understand this change. Can you provide an example?

Reply:
It handles this case:
```
#$ header function high_valuedarg_1(int, (int)(int))
def high_valuedarg_1(a, function=f1):
    x = function(a)
    return x
```

The default argument is a function so without the correction we get errors like:
```
E     File "/home/runner/work/pyccel/pyccel/tests/epyccel/modules/__epyccel__/highorder_functions_3ukrf85c09cx_3ukrf85c09cx.py", line 49
E       def high_valuedarg_1(a, function = def f1(a):
E                                            ^
E   SyntaxError: invalid syntax
```
(see https://github.com/pyccel/pyccel/runs/4056643786?check_suite_focus=true)

Reply:
I see, thanks!